rotkehlchen/chain/evm/decoding/balancer/utils.py

```diff
@@ -0,0 +1,223 @@
+            )
+            continue
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_unique_cache_value_at_ts(
+            write_cursor=write_cursor,
+            key_parts=cache_key_parts,
+            value=str(latest_pools_count),
+            timestamp=now,
+        )
+
+
+def _query_balancer_pools_count(chain: 'ChainID', version: Literal[1, 2]) -> int:
+    """Fetch the total number of pools for the specified chain and protoco
+    May raise:
+        - RemoteError
+        - DeserializationError
+    """
+    try:
+        response = requests.post(
+            url=BALANCER_API_URL,
+            json={
+                'variables': {
+                    'chain': CHAIN_ID_TO_BALANCER_API_MAPPINGS[chain],
+                    'version': version,
+                },
+                'query': GET_POOLS_COUNT_QUERY,
+            },
+            timeout=CachedSettings().get_timeout_tuple(),
+        )
+        response.raise_for_status()
+        data = response.json()
+    except (requests.exceptions.RequestException, JSONDecodeError) as e:
+        raise RemoteError(f'Balancer pools count query failed due to {e!s}') from e
+
+    return data['data']['poolGetPoolsCount']
+
+
+def _query_balancer_pools(
+        chain: 'ChainID',
+        database: 'DBHandler',
+        version: Literal[1, 2],
+) -> list[dict[str, Any]]:
+    """Fetches and processes Balancer pool data from API, storing token info in database.
+    Returns list of pool addresses. Supports V1 and V2 Balancer protocols.
+
+    May raise:
+        - RemoteError
+    """
+
+    all_pools: list[dict[str, Any]] = []
+    skip = 0
+    while True:
+        try:
+            response = requests.post(
+                url=BALANCER_API_URL,
+                json={
+                    'variables': {
+                        'skip': skip,
+                        'version': version,
+                        'first': BALANCER_API_CHUNK_SIZE,
+                        'chain': CHAIN_ID_TO_BALANCER_API_MAPPINGS[chain],
+                    },
+                    'query': GET_POOLS_QUERY,
+                },
+                timeout=CachedSettings().get_timeout_tuple(),
+            )
+            response.raise_for_status()
+            data = response.json()
+        except (requests.exceptions.RequestException, JSONDecodeError) as e:
+            raise RemoteError(f'Balancer pools query failed due to {e!s}') from e
+
+        if 'errors' in data:
+            raise RemoteError(f"Balancer pools query failed: {data['errors']}")
+
+        pools_chunk = data['data']['poolGetPools']
+        all_pools.extend(pools_chunk)
+
+        if len(pools_chunk) < BALANCER_API_CHUNK_SIZE:
+            break
+
+        skip += BALANCER_API_CHUNK_SIZE
+
+    return all_pools
+
+
+def _create_or_update_pool_token(
+        db: 'DBHandler',
+        chain_id: 'ChainID',
+        protocol: str,
+        pool: dict,
+) -> 'EvmToken':
+    underlying_tokens = [
```

Comment:
in this function you have several key Errors. Also can you use `deserialize_int` for the decimals?